src/v/resource_mgmt/memory_groups.cc

```diff
@@ -42,13 +42,15 @@ struct memory_shares {
     constexpr static size_t rpc = 20;
     constexpr static size_t recovery = 10;
     constexpr static size_t tiered_storage = 10;
+    constexpr static size_t admin = 5;
```

Comment:
Feels a bit heavy to take 5 shares away for the admin server. Does it really deserve half the same amount of memory that data_transforms or datalake uses?

I wonder if we should just share memory with the kafka server memory semaphore...